src/plugins/data/public/search/expressions/esdsl.test.ts

```diff
@@ -0,0 +1,128 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { esdsl } from './esdsl';
+
+jest.mock('@kbn/i18n', () => {
+  return {
+    i18n: {
+      translate: (id: string, { defaultMessage }: { defaultMessage: string }) => defaultMessage,
+    },
+  };
+});
+
+jest.mock('../../services', () => ({
+  getUiSettings: () => ({
+    get: () => true,
+  }),
+  getSearchService: () => ({
+    search: jest.fn((params: any) => {
+      return {
+        toPromise: async () => {
+          return { rawResponse: params };
+        },
+      };
+    }),
+  }),
+}));
+
+describe('esdsl', () => {
+  describe('correctly handles input', () => {
+    test('adds filters', async () => {
+      const result = await esdsl().fn(
+        {
+          type: 'kibana_context',
+          filters: [
+            {
+              meta: { index: '1', alias: 'test', negate: false, disabled: false },
+              query: { match_phrase: { gender: 'male' } },
+            },
+          ],
+        },
+        { dsl: '{}' },
+        { inspectorAdapters: {} } as any
+      );
+
+      expect(result).toMatchSnapshot();
+    });
+
+    test('adds filters to query with filters', async () => {
+      const result = await esdsl().fn(
+        {
+          type: 'kibana_context',
+          filters: [
+            {
+              meta: { index: '1', alias: 'test', negate: false, disabled: false },
+              query: { match_phrase: { gender: 'male' } },
+            },
+          ],
+        },
+        {
+          dsl:
+            '{"index": "kibana_sample_data_logs", "size": 4, "body": { "_source": false, "query": { "term": { "machine.os.keyword": "osx"}}}}',
+        },
+        { inspectorAdapters: {} } as any
+      );
+
+      expect(result).toMatchSnapshot();
+    });
+
+    test('adds query', async () => {
+      const result = await esdsl().fn(
+        {
+          type: 'kibana_context',
+          query: { language: 'lucene', query: '*' },
+        },
+        { dsl: '{}' },
+        { inspectorAdapters: {} } as any
+      );
+
+      expect(result).toMatchSnapshot();
+    });
+
+    test('adds query to a query with filters', async () => {
+      const result = await esdsl().fn(
+        {
+          type: 'kibana_context',
+          query: { language: 'lucene', query: '*' },
+        },
+        {
+          dsl:
+            '{"index": "kibana_sample_data_logs", "size": 4, "body": { "_source": false, "query": { "term": { "machine.os.keyword": "osx"}}}}',
+        },
+        { inspectorAdapters: {} } as any
+      );
+
+      expect(result).toMatchSnapshot();
+    });
+
+    test('ignores timerange', async () => {
+      const result = await esdsl().fn(
+        {
+          type: 'kibana_context',
+          timeRange: { from: 'now-15m', to: 'now' },
+        },
+        { dsl: '{}' },
+        { inspectorAdapters: {} } as any
+      );
+
+      expect(result).toMatchSnapshot();
+    });
+  });
```

Comment:
These tests look incomplete to me:

* What happens if the JSON is invalid?
* What if the JSON has line breaks?
* What if the context includes query, filters, and timerange?
* What if the JSON includes a `bool` query, how is that query merged with the context?